istioctl/cmd/sidecar.go

```diff
@@ -0,0 +1,291 @@
+				return fmt.Errorf("expecting an output filename")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			if err := readWorkloadGroup(filename, wg); err != nil {
+				return err
+			}
+			// consider passed in flags before attempting to infer config
+			cluster, err := gkeCluster(gkeProject, clusterLocation, clusterName)
+			if err != nil {
+				gkeProject, clusterLocation, clusterName, cluster, err = gkeConfig()
+				if err != nil {
+					return fmt.Errorf("could not infer (project, location, cluster). Try passing in flags instead")
+				}
+			}
+
+			if err = createConfig(wg, cluster, outputName); err != nil {
+				return err
+			}
+			fmt.Printf("generated config for (%s, %s, %s) in %s\n", gkeProject, clusterLocation, clusterName, outputName)
+			return nil
+		},
+	}
+	generateConfigCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	generateConfigCmd.PersistentFlags().StringVarP(&outputName, "output", "o", "", "Name of the tarball to be created")
+
+	generateConfigCmd.PersistentFlags().StringVar(&gkeProject, "project", "", "Target project name")
+	generateConfigCmd.PersistentFlags().StringVar(&clusterLocation, "location", "", "Target cluster location")
+	generateConfigCmd.PersistentFlags().StringVar(&clusterName, "cluster", "", "Target cluster name")
+	return generateConfigCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(wg *clientv1alpha3.WorkloadGroup, cluster *containerpb.Cluster, outputName string) error {
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	// TODO: pack temp into a tarball and then delete folder
+	// defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, cluster, temp); err != nil {
+		return err
+	}
+	if err = createHosts(temp); err != nil {
+		return err
+	}
+	if err = createCertificates(temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(wg, temp); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, cluster *containerpb.Cluster, dir string) error {
+	for _, v := range wg.Spec.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  cluster.ServicesIpv4Cidr,
```

Comment:
is all of this GKE stuff just for this? IMO we can default to `*` (match Pods) and then allow users to explicitly configure standard annotations like https://istio.io/latest/docs/reference/config/annotations/ `traffic.sidecar.istio.io/includeOutboundIPRanges`, to match what pods are doing

Reply:
Defaulting to `*` probably is more straightforward. Our current VM docs _should_ be updated to match (as this code matches the current instructions).